src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/KeyedStack.cs

```diff
@@ -3,28 +3,30 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Linq.Expressions.Compiler
 {
     /// <summary>
     /// A simple dictionary of stacks, keyed off a particular type
     /// This is useful for storing free lists of variables
     /// </summary>
-    internal sealed class KeyedStack<TKey, TValue> where TValue : class
+    internal sealed class KeyedStack<TKey, TValue> where TValue : class where TKey : notnull
     {
         private readonly Dictionary<TKey, Stack<TValue>> _data = new Dictionary<TKey, Stack<TValue>>();
 
         internal void Push(TKey key, TValue value)
         {
-            if (!_data.TryGetValue(key, out Stack<TValue> stack))
+            if (!_data.TryGetValue(key, out Stack<TValue>? stack))
             {
                 _data.Add(key, stack = new Stack<TValue>());
             }
 
             stack.Push(value);
         }
 
+        [return: MaybeNull]
         internal TValue TryPop(TKey key) =>
```

Comment:
Can be `internal TValue? TryPop(...)` because `TValue : class`.

Reply:
Yes